onnxruntime/core/providers/dnnl/dnnl_common.h

```diff
@@ -65,9 +66,14 @@ class PrimitivePool {
  private:
   // For thread safety, the map needs to be kept in thread local storage.
   static inline std::unordered_map<std::string, std::unique_ptr<PrimitiveBase>>& GetMap() {
-    static thread_local std::unordered_map<std::string, std::unique_ptr<PrimitiveBase>> map;
-    return map;
+    static thread_local auto map = std::make_unique<std::unordered_map<std::string, std::unique_ptr<PrimitiveBase>>>();
+    static thread_local RunOnUnload run_on_unload{
+        [&map = map]() {
+          map.reset();
```

Comment:
But the map is thread local, it may already gone at this point?

Reply:
If it is, then it's fine to reset it again, isn't it?

Reply:
But the unique_ptr itself is already destroyed?

Reply:
If the RunOnUnload thread_local gets destroyed, the callback won't run. So there should be no way to delete the already deleted unique_ptr.